src/context.rs

```diff
@@ -402,15 +408,24 @@ impl Context {
         let ret = Error::from_tss_rc(ret);
 
         if ret.is_success() {
-            unsafe {
-                let _ = MBox::from_raw(outpublic);
-                let _ = MBox::from_raw(creation_data);
-                let _ = MBox::from_raw(creation_hash);
-                let _ = MBox::from_raw(creation_ticket);
-            }
+            let out_public = unsafe { MBox::from_raw(outpublic) };
+            let creation_data = unsafe { MBox::from_raw(creation_data) };
+            let creation_hash = unsafe { MBox::from_raw(creation_hash) };
+            let creation_ticket = unsafe { MBox::from_raw(creation_ticket) };
+
+            let creation_data = CreationData::try_from(*creation_data)?;
+            let creation_hash = Digest::try_from(*creation_hash)?;
+            let creation_ticket = CreationTicket::try_from(*creation_ticket)?;
+
             let primary_key_handle = KeyHandle::from(esys_prim_key_handle);
             let _ = self.open_handles.insert(primary_key_handle.into());
-            Ok(primary_key_handle)
+            Ok((
+                primary_key_handle,
+                *out_public,
+                creation_data,
+                creation_hash,
+                creation_ticket,
+            ))
```

Comment:
Really like that all the data is returned it is something I have been missing!

Though I am not to found of how it is returned. I would much more have liked it if there were a struct that was returned.